java-checks/src/test/files/checks/StaticFieldInitializationCheck.java

```diff
@@ -55,6 +55,17 @@ public static int getInstance5() {
     instance5 = 12;  // compliant - primitive type
     return instance5;
   }
+
+  protected static Object instance6 = null;
```

Comment:
arguable : I don't see the testcase which will make you require the stack.

Reply:
Classes B,C, and D from test files are nested classes of A, which has a synchronized method.